src/librustc_codegen_llvm/builder.rs

```diff
@@ -1068,34 +1069,17 @@ impl StaticBuilderMethods for Builder<'a, 'll, 'tcx> {
         self.cx().get_static(def_id)
     }
 
-    fn static_panic_msg(
-        &mut self,
-        msg: Option<Symbol>,
-        filename: Symbol,
-        line: Self::Value,
-        col: Self::Value,
-        kind: &str,
-    ) -> Self::Value {
+    fn static_panic_location(&mut self, loc: &syntax::source_map::Loc) -> Self::Value {
+        let filename = Symbol::intern(&loc.file.name.to_string());
+        let filename = self.const_str(filename);
+        let line = self.const_u32(loc.line as u32);
+        let col = self.const_u32(loc.col.to_usize() as u32 + 1);
+        let struct_ = self.const_struct(&[filename.0, filename.1, line, col], false);
+
         let align = self.tcx.data_layout.aggregate_align.abi
             .max(self.tcx.data_layout.i32_align.abi)
             .max(self.tcx.data_layout.pointer_align.abi);
-
-        let filename = self.const_str_slice(filename);
-
-        let with_msg_components;
-        let without_msg_components;
-
-        let components = if let Some(msg) = msg {
-            let msg = self.const_str_slice(msg);
-            with_msg_components = [msg, filename, line, col];
-            &with_msg_components as &[_]
-        } else {
-            without_msg_components = [filename, line, col];
-            &without_msg_components as &[_]
-        };
-
-        let struct_ = self.const_struct(&components, false);
-        self.static_addr_of(struct_, align, Some(kind))
+        self.static_addr_of(struct_, align, None)
```

Comment:
You can add a FIXME for moving this into miri (which is better equipped to do this correctly even if e.g. the order of the fields changes). cc @oli-obk